src/main/java/org/spongepowered/api/data/context/DataContextual.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.data.context;
+
+import org.spongepowered.api.context.ContextViewer;
+import org.spongepowered.api.context.NamedContextual;
+
+/**
+ * An object that has per-{@link ContextViewer} views of data.
+ */
+public interface DataContextual extends NamedContextual {
+
+    /**
+     * Tests if the provided viewer has an active context.
+     *
+     * @param viewer The viewer
+     * @return {@code true} if the viewer has a context
+     */
+    boolean hasContext(ContextViewer viewer);
+
+    /**
+     * Gets the context as seen by the viewer.
+     *
+     * @param viewer The viewer of the context
+     * @return The context
+     */
+    DataContext getContext(ContextViewer viewer);
```

Comment:
Potentially instead create a getOrCreateContext, and have this one return an optional?